### ISSUE TITLE
Setting a GUC in single-user mode segfaults

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#3172]
```
$ bin/postmaster --single -D data-master/ postgres

PostgreSQL stand-alone backend 8.3.23
backend> set enable_seqscan=off;
�w@���wbtct����������������

                               �o�V�[�V���Z���z�VXbz�V}az�V�Wz�VV2z�V��q�V��q�V�(\�V�[�V`�[�V��[�VYg[�VSegmentation fault (core dumped)
```

**Comment by hlinnaka** [comment]
```
(gdb) bt
#0  raise (sig=11) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00005610a66fcd53 in StandardHandlerForSigillSigsegvSigbus_OnMainThread (processName=0x5610a6a90873 "Master process", postgres_signal_arg=11) at elog.c:4625
#2  0x00005610a65b970f in CdbProgramErrorHandler (postgres_signal_arg=11) at postgres.c:3609
#3  <signal handler called>
#4  0x00005610a679bda8 in cdbconn_doConnectStart (segdbDesc=0x5610a82f9710, gpqeid=0x7ffd699bff50 "1;0;558012332052934;true;1;4", 
    options=0x5610a82fbe78 " -c gp_qd_hostname=127.0.0.1 -c gp_qd_port=5432 -c client_min_messages=notice -c codegen=false -c codegen_advance_aggregate=false -c codegen_exec_eval_expr=false -c codegen_exec_variable_list=false -c"...) at cdbconn.c:475
#5  0x00005610a67aaae5 in createGang_async (type=GANGTYPE_PRIMARY_WRITER, gang_id=1, size=3, content=-1) at cdbgang_async.c:127
#6  0x00005610a67a6258 in createGang (type=GANGTYPE_PRIMARY_WRITER, gang_id=1, size=3, content=-1) at cdbgang.c:292
#7  0x00005610a67a617d in AllocateWriterGang () at cdbgang.c:263
#8  0x00005610a67a57ee in cdbdisp_dispatchSetCommandToAllGangs (strCommand=0x5610a829d440 "SET enable_seqscan TO 'off'", serializedQuerytree=0x0, serializedQuerytreelen=0, 
    serializedPlantree=0x0, serializedPlantreelen=0, cancelOnError=0 '\000', needTwoPhase=0 '\000', ds=0x7ffd699c0310) at cdbdisp_query.c:1446
#9  0x00005610a67a3256 in CdbDispatchSetCommand (strCommand=0x5610a829d440 "SET enable_seqscan TO 'off'", cancelOnError=0 '\000', needTwoPhase=0 '\000') at cdbdisp_query.c:307
#10 0x00005610a671aebe in DispatchSetPGVariable (name=0x5610a82f3f00 "enable_seqscan", args=0x5610a82f4008, is_local=0 '\000') at guc.c:5665
#11 0x00005610a671a78b in ExecSetVariableStmt (stmt=0x5610a82f4060) at guc.c:5462
#12 0x00005610a65c28f8 in ProcessUtility (parsetree=0x5610a82f4060, queryString=0x5610a82f35c0 "set enable_seqscan=off;\n", params=0x0, isTopLevel=1 '\001', 
    dest=0x5610a6e688c0 <debugtupDR>, completionTag=0x7ffd699c0830 "") at utility.c:1564
#13 0x00005610a65bf105 in PortalRunUtility (portal=0x5610a82f7600, utilityStmt=0x5610a82f4060, isTopLevel=1 '\001', dest=0x5610a6e688c0 <debugtupDR>, completionTag=0x7ffd699c0830 "")
    at pquery.c:1382
#14 0x00005610a65bf360 in PortalRunMulti (portal=0x5610a82f7600, isTopLevel=1 '\001', dest=0x5610a6e688c0 <debugtupDR>, altdest=0x5610a6e688c0 <debugtupDR>, completionTag=0x7ffd699c0830 "")
    at pquery.c:1475
#15 0x00005610a65be7f8 in PortalRun (portal=0x5610a82f7600, count=9223372036854775807, isTopLevel=1 '\001', dest=0x5610a6e688c0 <debugtupDR>, altdest=0x5610a6e688c0 <debugtupDR>, 
    completionTag=0x7ffd699c0830 "") at pquery.c:1013
#16 0x00005610a65b6759 in exec_simple_query (query_string=0x5610a82f35c0 "set enable_seqscan=off;\n", seqServerHost=0x0, seqServerPort=-1) at postgres.c:1811
#17 0x00005610a65bb5ed in PostgresMain (argc=5, argv=0x5610a804c200, dbname=0x5610a801bc70 "postgres", username=0x5610a80169b0 "heikki") at postgres.c:5004
#18 0x00005610a6455351 in main (argc=5, argv=0x5610a804c200) at main.c:198
```

**Comment by hlinnaka** [comment]
Fixed by commit 814d82d682.
